### PR TITLE
🏗♻️🚀 Updates to `app-index` tests

### PR DESCRIPTION
**Pull request by alanorozco**
In general, makes the dev dashboard tests faster and more consistent with other `build-system` tests.

1. **Removes AMP validation of output**. This was the slowest process in tests. It wasn't providing much value since we'd never fail validation anyway. We could potentially use Bento components later so might as well remove.

2. **Uses `ava`**. This is faster and can be listed along other `ava` tests instead of requiring a custom runner. Also removed special runner, task, and related configuration.

3. **Uses `posthtml` instead of `JSDOM`**. This reduces the elapsed time of remaining tests from `~1s` to `~200ms`.

### CODE REVIEW

File: build-system/pr-check/build-targets.js [hunk 1]
```diff
@@ -173,16 +173,6 @@ const targetMatchers = {
...
     );
   },
-  [Targets.DEV_DASHBOARD]: (file) => {
-    if (isOwnersFile(file)) {
-      return false;
-    }
-    return (
-      file == 'build-system/tasks/dev-dashboard-tests.js' ||
-      file == 'build-system/server/app.js' ||
-      file.startsWith('build-system/server/app-index/')
```
Comment (alanorozco) [inline build-system/pr-check/build-targets.js:183]: No need to add new targets since `Targets.AVA` matches `build-system/server/*` already.

**Review by rsimha — APPROVED** [review]
👏👏👏

**Comment by rcebulko** [comment]
This eliminated the need for some allowlist entries: 
image

Will include fix in https://github.com/ampproject/amphtml/pull/34877

**Comment by rileyajones** [comment]
#34877 looks like it might take a while to land. Lets get this fixed quickly so we can get CI green again #34893

**Pull request merged**